Dockerfile

```diff
@@ -0,0 +1,49 @@
+# This file is used for building and testing agent.
+# It includes some useful tools for development of agent
+# which can ensure everyone using similar development toolkit
+
+FROM centos:7
+
+ARG http_proxy
+ARG https_proxy
+
+# TODO: if you have trouble downloading git repos due to cert problem
+# try to uncomment this
+# ENV GIT_SSL_NO_VERIFY true
+
+# install building tools
+RUN yum makecache && yum install -y \
+	git automake libtool glibc-headers gcc-c++ make
+
+# install GO development environment
+ENV GO_VERSION 1.9.2
+RUN curl -fkL https://storage.googleapis.com/golang/go${GO_VERSION}.linux-amd64.tar.gz \
+	| tar -zxC /usr/local/
+ENV GOPATH /go
+ENV PATH $PATH:/go/bin:/usr/local/go/bin
+
+# install golang/protobuf
+ENV PROTOBUF_COMMIT 1e59b77b52bf8e4b449a57e6f79f21226d571845
+RUN curl -fkL https://github.com/google/protobuf/releases/download/v3.5.0/protobuf-cpp-3.5.0.tar.gz \
+	| tar -zxC /opt && cd /opt/protobuf-3.5.0 \
+	&& ./autogen.sh && ./configure && make && make install
+RUN go get -d github.com/golang/protobuf/protoc-gen-go \
+	&& cd $GOPATH/src/github.com/golang/protobuf/ \
+	&& git checkout -q ${PROTOBUF_COMMIT} \
+	&& go install github.com/golang/protobuf/protoc-gen-go
+
+# install gogo/protobuf
+ENV GOGO_COMMIT 41168f6614b7bb144818ec8967b8c702705df564
```

Comment:
Gotta love the [Go-go's](https://en.wikipedia.org/wiki/The_Go-Go%27s) :smile:

Reply:
😆